pyrex.py

```diff
@@ -316,11 +316,14 @@ def get_engine_info(config):
 
 
 def get_subid_length(filename, name):
+    r = re.compile(r'{}:\d+:(\d+)'.format(re.escape(name)))
+
     with open(filename, "r") as f:
         for line in f:
-            (ident, _, id_length) = line.rstrip().split(":")
-            if ident == name:
-                return int(id_length)
+            m = r.search(line)
```

Comment:
`r.search()` is not anchored to the beginning of the line like `r.match()` is. Either use r.match() or manually anchor the expression to the beginning of the line.

For example, your code would currently match the first line, when we want the second:
```
# foo:1:2
foo:3:4
```